lib/puma/dsl.rb

```diff
@@ -731,6 +729,8 @@ def on_booted(&block)
     # @version 5.0.0
     #
     def on_refork(key = nil, &block)
+      warn_if_in_single_mode('on_refork')
```

Comment:
I think this makes sense here? This seemed to be the only cluster mode related hook that was missing the warning.

Reply:
We should probably warn for the `fork_worker` config option too.

Reply:
Looking at the log, I think it was only intended to be called in hook methods: https://github.com/puma/puma/pull/3376/files#diff-122c03b9c96f060a9cc29b5ed3446467e1909b0054b4f5cfd95ee41cfff832e7R1408-R1421

Also, there are other non-hook cluster-only methods that we don't warn on (`preload_app!`, `worker_timeout`, `worker_boot_timeout` etc.). I'm happy to open a separate PR to tackle those (including `fork_worker`) with a separate log, or maybe even consolidate them.

Reply:
> I'm happy to open a separate PR to tackle those (including fork_worker) with a separate log, or maybe even consolidate them.

Sure, we can consider this a separate issue.